src/marqo/tensor_search/add_docs.py

```
@@ -64,119 +82,185 @@ def threaded_download_and_preprocess_images(
+                            except OSError as e:
+                                if "image file is truncated" in str(e):
+                                    media_repo[doc[field]] = e
+                                    metric_obj.increment_counter(f"{doc.get(field, '')}.OSError")
+                                    continue
+                                else:
+                                    raise e
+
+                    elif (modality in [Modality.VIDEO,
+                                       Modality.AUDIO] and marqo_index.type == IndexType.Unstructured) or (
+                            marqo_index.type == IndexType.Structured and media_field_types_mapping[field] in 
+                            [FieldType.AudioPointer, FieldType.VideoPointer]):
+                        if marqo_index.model.properties.get('type') not in [
+                            ModelType.LanguageBind] and modality not in marqo_index.model.properties.get(
+                            'supported_modalities'):
+                            media_repo[doc[field]] = UnsupportedModalityError(
+                                f"Model {marqo_index.model.name} does not support {modality}")
+                            continue
+                        try:
+                            processed_chunks = download_and_chunk_media(doc[field], device, download_headers, modality,
+                                                                        marqo_index, preprocessors)
+                            media_repo[doc[field]] = processed_chunks
+                        except (ffmpeg.Error, S2InferenceError) as e:
+                            logger.error(f"Error processing {modality} file: {str(e)}")
+                            media_repo[doc[field]] = S2InferenceError(f"Error processing {modality} file: {str(e)}")
+
+                    else:
+                        pass
```

Should we raise an error here?

---

done

---

All good. A key error (which goes to a 500) is not a bad idea in this case.